server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -86,34 +89,32 @@ public class GarbageCollectWriteAheadLogs {
     this.fs = fs;
     this.useTrash = useTrash;
     this.liveServers = liveServers;
-    this.walMarker = new WalStateManager(context);
-    this.store = () -> Iterators.concat(
-        TabletStateStore.getStoreForLevel(DataLevel.ROOT, context).iterator(),
-        TabletStateStore.getStoreForLevel(DataLevel.METADATA, context).iterator(),
-        TabletStateStore.getStoreForLevel(DataLevel.USER, context).iterator());
+    this.walMarker = createWalStateManager(context);
+    this.hasCollected = new AtomicBoolean(false);
   }
 
-  /**
-   * Creates a new GC WAL object. Meant for testing -- allows mocked objects.
-   *
-   * @param context the collection server's context
-   * @param fs volume manager to use
-   * @param useTrash true to move files to trash rather than delete them
-   * @param liveTServerSet a started LiveTServerSet instance
-   */
   @VisibleForTesting
-  GarbageCollectWriteAheadLogs(ServerContext context, VolumeManager fs, boolean useTrash,
-      LiveTServerSet liveTServerSet, WalStateManager walMarker,
-      Iterable<TabletLocationState> store) {
-    this.context = context;
-    this.fs = fs;
-    this.useTrash = useTrash;
-    this.liveServers = liveTServerSet;
-    this.walMarker = walMarker;
-    this.store = store;
+  WalStateManager createWalStateManager(ServerContext context) {
```

Comment:
I may be misunderstanding but what is the point of these new `createWalStateManager` and `createStore` methods? In the testing code, I don't see what the benefit of overriding these is. For example, the `createWalStateManager` is overridden to return marker. How is this any different from just simply passing marker to the constructor? Is the point to just get rid of the one `GarbageCollectWriteAheadLogs` constructor that is meant to be used for testing?

This could very well be the right thing to do, I just was a bit confused looking at it

Reply:
One option to clean this up would be to have a package-private constructor that is visible for testing that accepts a pre-constructed store and WalStateManager. This constructor for normal use case would just be a pass-through to that other constructor, but the other constructor could be reused by test code that provides mock versions of these objects. That might be easier to grok than trying to override these methods with custom implementations and using the same constructor for both regular use and test use.

Reply:
> I may be misunderstanding but what is the point of these new `createWalStateManager` and `createStore` methods?

The point it to make it easier possible (or at least easier) to use mock objects for testing. This lets us override these methods and return the mock objects.